producer.go

```diff
@@ -342,12 +342,21 @@ func (w *Producer) popTransaction(frameType int32, data []byte) {
 }
 
 func (w *Producer) transactionCleanup() {
-	// clean up transactions we can easily account for
-	for _, t := range w.transactions {
-		t.Error = ErrNotConnected
-		t.finish()
+l:
```

Comment:
I don't _think_ this label is necessary since it's not a nested `for` loop.

Reply:
In ```for``` loop, there's a ```select``` in it, without label, ```break``` only jump out from ```select``` not ```for```.

Reply:
In ```for``` loop, there's a ```select``` in it, without label, ```break``` only jump out from ```select``` not ```for```.